tokio/src/sync/mpsc/bounded.rs

```diff
@@ -150,6 +152,44 @@ impl<T> Receiver<T> {
         poll_fn(|cx| self.poll_recv(cx)).await
     }
 
+    /// Blocking receive to call outside of asynchronous contexts
+    ///
+    /// ```
+    /// use std::thread;
+    /// use tokio::runtime::Runtime;
+    /// use tokio::sync::mpsc;
+    ///
+    /// fn main() {
+    ///     let (mut tx, mut rx) = mpsc::channel::<u8>(10);
+    ///     
+    ///     let sync_code = thread::spawn(move || {
+    ///         assert_eq!(Ok(Some(10)), rx.blocking_recv());
+    ///     });
+    ///
+    ///     Runtime::new()
+    ///         .unwrap()
+    ///         .block_on(async move {
+    ///             let _ = tx.send(10).await;
+    ///         });
+    ///     sync_code.join().unwrap()
+    /// }
+    /// ```
+    pub fn blocking_recv(&mut self) -> Result<Option<T>, crate::park::ParkError> {
+        if Handle::try_current().is_ok() {
+            panic!("blocking_recv cannot be called from within a runtime");
+        }
+        let mut park = CachedParkThread::new();
+        let waker = park.get_unpark()?.into_waker();
+        let mut cx = Context::from_waker(&waker);
+
+        loop {
+            if let Poll::Ready(v) = crate::coop::budget(|| self.poll_recv(&mut cx)) {
+                return Ok(v);
+            }
```

Comment:
The coop budget is not necessary for this.

Reply:
Done :+1: